### DISCUSSION TITLE
Mimicking the modular structure of this app

### DISCUSSION THREAD

**Discussion by j-j-m** [pointfreeco/isowords#55] [Q&A] [answered]
Hi. Love the project and the way it is structured. I have tried to move one of my own apps over to a similar sort of scheme by following the setup of your package file

I am seeing a few flaky CI runs, sometimes testing passes and sometimes it fails. 

the failure seems to be related to the resolution of XCTestDynamicOverlay
```
error: Missing package product 'XCTestDynamicOverlay' 
```

when I see this locally I can get rid of the issue by clearing my derived data which then prompts Xcode to do its thing and pull the dependencies again.

I know it may be difficult to diagnose without having a copy of my project, but I figure it was worth an ask. Maybe you guys have seen something similar? I am using an almost identical copy of your `ci` workflow and a paired down set of package dependencies. Hopefully I am missing something small.

**Comment by stephencelis** [comment, marked as answer]
@j-j-m That could be related to an SPM bug where dependency URLs diverge, _e.g._, if the `.git` is present or not:

```diff
-https://github.com/pointfreeco/xctest-dynamic-overlay
+https://github.com/pointfreeco/xctest-dynamic-overlay.git
```

We saw the same CI failures here. The key is to depend on `xctest-dynamic-overlay` in the same way it is depended on in any upstream packages. In TCA and `combine-schedulers` this is _without_ the trailing `.git`. Hopefully this issue will be resolved in a future version of Swift/Xcode.

**Comment by j-j-m** [comment]
thanks for the quick reply.
